### PR TITLE
1.27 Backport H/2 header discard fix

### PR DESCRIPTION
**Pull request by SeanKilleen**
Commit Message:
Discard the Host header if the :authority header was received to bring Envoy into compliance with https://www.rfc-editor.org/rfc/rfc9113#section-8.3.1

This behavioral change can be reverted by setting runtime flag envoy.reloadable_features.http2_discard_host_header to false.

Risk Level: Low
Testing: Unit Tests
Docs Changes: N/A
Release Notes: Yes
Platform Specific Features: N/A
Runtime guard: envoy.reloadable_features.http2_discard_host_header
Back-ports #30005, supporting #31118

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -2,6 +2,35 @@ date: Pending
...
+- area: stats dns_filter
+  change: |
+    Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.udp.dns_filter.v3.DnsFilterConfig>`
+    is properly extracted. This changes the Prometheus name from
+    dns_filter_myprefix_local_a_record_answers{} to dns_filter_local_a_record_answers{envoy.dns_filter_prefix="myprefix"}.
+- area: http2
+  change: |
+    Changes the default value of ``envoy.reloadable_features.http2_use_oghttp2`` to true. This changes the codec used for HTTP/2
+    requests and responses. This behavior can be reverted by setting the feature to false.
+- area: http2
```
Comment (SeanKilleen) [inline changelogs/current.yaml:29]: ⚠️ Self-review -- This is the only note that the back-ported change addresses. I'm not sure if the other notes should be included here or not as part of the process.

Comment (phlax) [reply changelogs/current.yaml:29]: yeah just the change from the original commit

**Comment by SeanKilleen** [comment]
Not sure what happened here; was following pretty explicit steps. I'll get it together and try once more.

**Comment by phlax** [comment]
ping me on slack if you want - happy to help - would be good to get these backports on the branch ahead of next release

**Comment by phlax** [comment]
probably the step i missed out was fetching from envoys current branches

**Pull request closed without merging**